### ISSUE TITLE
Cannot Save file and Export as PDF

### ISSUE THREAD

**Issue by chanemaster** [xournalpp/xournalpp#3145]
**Affects versions :**
 - OS: Windows 10
 - Which version of libgtk do you use: 3.24.29
 - Version of Xournal++: 1.1.0+dev
 - Installation method: from Automated nightly build webpage which is "xournalpp-1.1.0.dev-nightly.20210602-windows.zip"

**Describe the bug**
Cannot save file: The error message below shows everytime after I click File > Save, File > Export as PDF (also File > Save as, File > Export as...). So I am not be able to save any file at all.

> The folder contents could not be displayed
> 
> Error when getting information for file
> 
> “C:\Program Files\Xournal++\file:\C:\Program%20Files\Xournal++”
> 
> : Input/output error

**To Reproduce**
1. Click File
2. Click Save or Export as PDF or Save as or Export as...

**Expected behavior**

**Screenshots of Problem**
(Optional) If applicable, add screenshots to help explain your problem.

![Export as pdf problem2](https://user-images.githubusercontent.com/85364302/120936621-f523fa00-c732-11eb-91cc-a09418b09900.PNG)

**Additional context**
Add any other information or context about the problem here.
Can be read at https://github.com/xournalpp/xournalpp/discussions/3144

**Comment by rolandlo** [comment]
@mitzal @Febbe My first thought was, that this might be a regression due to #3046. But #3046 only changes the path of the `tex.tex` file, right? Any idea how a path like `C:\ProgramFiles\Xournal++\file:\C:\Program%20Files\Xournal++` occurs?

**Comment by Febbe** [comment]
Could be a missing "\n" in the error message, or two paths are concatenated. The thing which bothers me is, that a URI is tried to be used.

**Comment by Febbe** [comment]
@chanemaster Do you had an old version of Xournal++ installed before? It might help, to just delete the settings.
They are located under `C:\Users\<user>\AppData\Local\xournalpp`.

@rolandlo, just tried it, and it seems to be, that the first place which is opened is the installation dir, not the Documents folder.

**Comment by chanemaster** [comment]
Yes, I used to install an old version of Xournal++ but that version could not save as well.
I think this folder C:\Users\<user>\AppData\Local\xournalpp cannot be deleted since last time I changed the name of settings.xml and then my last settings in Xournal++ were lost and I had to set it again.

However, I have noticed that there is a folder name .xournalpp underC:\Users\<user> (this is before I click AppData Folder, please see the screenshot attached below).Please let me know if I should delete this or not.
After I delete it, will I still be able to use Xournal++ because I will need to use it in about 5 hours.
Best Regards,Charn



    On Tuesday, June 8, 2021, 01:22:36 PM GMT+7, Fabian Keßler ***@***.***> wrote:  
 
 


@chanemaster Do you had an old version of Xournal++ installed before? It might help, to just delete the settings.
They are located under C:\Users\<user>\AppData\Local\xournalpp.

@rolandlo, just tried it, and it seems to be, that the first place which is opened is the installation dir, not the Documents folder.

**Comment by rolandlo** [comment]
> However, I have noticed that there is a folder name .xournalpp underC:\Users\<user> (this is before I click AppData Folder, please see the screenshot attached below).Please let me know if I should delete this or not. After I delete it, will I still be able to use Xournal++ because I will need to use it in about 5 hours. Best Regards,Charn

The folder `.xournalpp` is left from a previous version indeed. This folder contains autosave files, errorlogs, metadata of opened files and the `settings.xml` file for the previous version. Unless you need some of it for a particular reason, you can safely delete it. I really don't believe that deleting it will solve the problem though. By the way, are any xopp-files saved in the autosave folder? That would suggest that autosaving works, but saving manually doesn't.

One thing that might indeed help is to manually set the `lastSavePath` property in the currently used `settings.xml` file. You can edit the `settings.xml` file with a simple text editor.


By the way, when you renamed your `settings.xml` file to try if it solved the problem (which it didn't), you could have simply deleted the freshly created `settings.xml` file and renamed the backup copy to `settings.xml` again.

**Comment by Febbe** [comment]
@rolandlo, deleting the old settings folder and renaming the new can indeed fix it, since we make a copy of the old.
@chanemaster, can you zip both and attach them here?

**Comment by chanemaster** [comment]
Thank you very much. After I delete folder .xournalpp, I can save now.Do you still want zip files to check so that I can send them to you?
Best Regards,Charn
    On Tuesday, June 8, 2021, 07:59:41 PM GMT+7, Fabian Keßler ***@***.***> wrote:  
 
 


@rolandlo, deleting the old settings folder and renaming the new can indeed fix it, since we make a copy of the old.
@chanemaster, can you zip both and attach them here?

**Comment by rolandlo** [comment]
> Thank you very much. After I delete folder .xournalpp, I can save now.Do you still want zip files to check so that I can send them to you? Best Regards,Charn

Wonderful. Yes, the zip-files would be still interesting to examine the cause of the problem. The `settings.xml` files should be enough.

**Comment by chanemaster** [comment]
I attached it for you.You mean this file, right?

    On Wednesday, June 9, 2021, 10:29:59 AM GMT+7, Roland Lötscher ***@***.***> wrote:  
 
 



Thank you very much. After I delete folder .xournalpp, I can save now.Do you still want zip files to check so that I can send them to you? Best Regards,Charn


Wonderful. Yes, the zip-files would be still interesting to examine the cause of the problem. The settings.xml files should be enough.

**Comment by rolandlo** [comment]
I don't see the file. How did you attach it? You can just drag and drop it into the reply.

**Comment by chanemaster** [comment]
I just attached and sent again.Please tell me if you don't see it so that I can send to you again.
    On Wednesday, June 9, 2021, 12:00:48 PM GMT+7, Roland Lötscher ***@***.***> wrote:  
 
 


I don't see the file. How did you attach it? You can just drag and drop it into the reply.

**Comment by rolandlo** [comment]
I still don't see the file. You can check in the Preview, if the file is attached.

**Comment by chanemaster** [comment]
I attached it as shown on screenshot. Wonder why it is not show to your computer.So this time I attached it and also clicked inside the file and copied its content to you below.Please let me know whether this should work.
<?xml version="1.0" encoding="UTF-8"?><settings>  <property name="pressureSensitivity" value="true"/>  <property name="minimumPressure" value="0.05000000"/>  <property name="pressureMultiplier" value="1.00000000"/>  <property name="zoomGesturesEnabled" value="true"/>  <property name="selectedToolbar" value="Portrait"/>  <property name="lastSavePath" value="C:\Users\charn\OneDrive\Desktop\Charming Edunicornâ„¢\#####MATH NOTE\Xournalpp Save"/>  <property name="lastOpenPath" value="C:\Users\charn\OneDrive\iMATHgine\Math N Boom"/>  <property name="lastImagePath" value="C:\Users\charn\OneDrive\Pictures\LINE Stickers(From iPad)"/>  <property name="zoomStep" value="10.00000000"/>  <property name="zoomStepScroll" value="2.00000000"/>  <property name="displayDpi" value="72"/>  <property name="mainWndWidth" value="1296"/>  <property name="mainWndHeight" value="877"/>  <property name="maximized" value="false"/>  <property name="showToolbar" value="true"/>  <property name="showSidebar" value="false"/>  <property name="sidebarWidth" value="50"/>  <property name="sidebarOnRight" value="false"/>  <property name="scrollbarOnLeft" value="false"/>  <property name="menubarVisible" value="true"/>  <property name="numColumns" value="1"/>  <property name="numRows" value="1"/>  <property name="viewFixedRows" value="false"/>  <property name="showPairedPages" value="false"/>  <property name="layoutVertical" value="false"/>  <property name="layoutRightToLeft" value="false"/>  <property name="layoutBottomToTop" value="false"/>  <property name="numPairsOffset" value="1"/>  <property name="presentationMode" value="false"/>    <property name="fullscreenHideElements" value="mainMenubar"/>    <property name="presentationHideElements" value="mainMenubar,sidebarContents"/>    <property name="stylusCursorType" value="big"/>  <property name="highlightPosition" value="true"/>  <property name="cursorHighlightColor" value="2149777152"/>  <property name="cursorHighlightBorderColor" value="2147483903"/>  <property name="cursorHighlightRadius" value="30.00000000"/>  <property name="cursorHighlightBorderWidth" value="0.00000000"/>  <property name="darkTheme" value="false"/>  <property name="disableScrollbarFadeout" value="false"/>    <property name="scrollbarHideType" value="none"/>  <property name="autoloadPdfXoj" value="true"/>  <property name="defaultSaveName" value="%F-Note-%H-%M"/>  <property name="autosaveEnabled" value="true"/>  <property name="autosaveTimeout" value="3"/>  <property name="addHorizontalSpace" value="false"/>  <property name="addHorizontalSpaceAmount" value="150"/>  <property name="addVerticalSpace" value="false"/>  <property name="addVerticalSpaceAmount" value="150"/>  <property name="drawDirModsEnabled" value="false"/>  <property name="drawDirModsRadius" value="50"/>  <property name="snapRotation" value="true"/>  <property name="snapRotationTolerance" value="0.20000000"/>  <property name="snapGrid" value="true"/>  <property name="snapGridTolerance" value="0.25000000"/>  <property name="snapGridSize" value="14.17000000"/>  <property name="touchDrawing" value="false"/>  <property name="pressureGuessing" value="false"/>  <property name="selectionBorderColor" value="4294967295"/>  <property name="backgroundColor" value="4294967295"/>  <property name="selectionMarkerColor" value="4285702095"/>  <property name="touchZoomStartThreshold" value="0.00000000"/>  <property name="pageRerenderThreshold" value="5.00000000"/>    <property name="pdfPageCacheSize" value="10"/>  <property name="preloadPagesBefore" value="3"/>  <property name="preloadPagesAfter" value="5"/>  <property name="eagerPageCleanup" value="true"/>    <property name="pageTemplate" value="xoj/template&#10;copyLastPageSize=false&#10;copyLastPageSettings=false&#10;size=841.889764x1190.551181&#10;backgroundType=plain&#10;backgroundColor=#ffffffff&#10;"/>  <property name="sizeUnit" value="cm"/>  <property name="audioFolder" value="file:///C:/Users/charn/OneDrive/Desktop"/>  <property name="audioInputDevice" value="1"/>  <property name="audioOutputDevice" value="5"/>  <property name="audioSampleRate" value="44100.00000000"/>  <property name="audioGain" value="1.00000000"/>  <property name="defaultSeekTime" value="5"/>  <property name="pluginEnabled" value=""/>  <property name="pluginDisabled" value=""/>  <property name="strokeFilterIgnoreTime" value="150"/>  <property name="strokeFilterIgnoreLength" value="1.00000000"/>  <property name="strokeFilterSuccessiveTime" value="500"/>  <property name="strokeFilterEnabled" value="false"/>  <property name="doActionOnStrokeFiltered" value="false"/>  <property name="trySelectOnStrokeFiltered" value="false"/>  <property name="snapRecognizedShapesEnabled" value="false"/>  <property name="restoreLineWidthEnabled" value="false"/>  <property name="numIgnoredStylusEvents" value="0"/>  <property name="inputSystemTPCButton" value="false"/>  <property name="inputSystemDrawOutsideWindow" value="true"/>  <property name="preferredLocale" value=""/>  <property name="stabilizerAveragingMethod" value="0"/>  <property name="stabilizerPreprocessor" value="0"/>  <property name="stabilizerBuffersize" value="20"/>  <property name="stabilizerSigma" value="0.50000000"/>  <property name="stabilizerDeadzoneRadius" value="1.30000000"/>  <property name="stabilizerDrag" value="0.40000000"/>  <property name="stabilizerMass" value="5.00000000"/>  <property name="stabilizerCuspDetection" value="true"/>  <property name="stabilizerFinalizeStroke" value="true"/>  <property name="latexSettings.autoCheckDependencies" value="true"/>  <property name="latexSettings.globalTemplatePath" value="C:\Users\charn\AppData\Local\MiKTeX\TeXworks\0.6\templates\Basic LaTeX documents\default_template.tex"/>  <property name="latexSettings.genCmd" value="pdflatex -halt-on-error -interaction=nonstopmode '{}'"/>  <property name="font" font="Layiji TaMaiTine1" size="20.00000000"/>  <data name="buttonConfig">    <data name="default">      <attribute name="color" type="hex" value="ff000000"/>      <attribute name="drawingType" type="string" value="default"/>      <attribute name="size" type="string" value="thin"/>      <attribute name="tool" type="string" value="pen"/>    </data>    <data name="eraser">      <attribute name="eraserMode" type="string" value="none"/>      <attribute name="size" type="string" value="none"/>      <attribute name="tool" type="string" value="eraser"/>    </data>    <data name="middle">      <attribute name="tool" type="string" value="none"/>    </data>    <data name="right">      <attribute name="tool" type="string" value="none"/>    </data>    <data name="stylus">      <attribute name="tool" type="string" value="none"/>    </data>    <data name="stylus2">      <attribute name="tool" type="string" value="none"/>    </data>    <data name="touch">      <attribute name="device" type="string" value=""/>      <attribute name="disableDrawing" type="boolean" value="false"/>      <attribute name="tool" type="string" value="none"/>    </data>  </data>  <data name="deviceClasses">    <data name="Pentablet  Eraser ">      <attribute name="deviceClass" type="int" value="2"/>      <attribute name="deviceSource" type="int" value="1"/>    </data>    <data name="Pentablet  Stylus Pen ">      <attribute name="deviceClass" type="int" value="2"/>      <attribute name="deviceSource" type="int" value="1"/>    </data>    <data name="System Aggregated Pointer">      <attribute name="deviceClass" type="int" value="1"/>      <attribute name="deviceSource" type="int" value="0"/>    </data>  </data>  <data name="tools">    <attribute name="current" type="string" value="image"/>    <data name="drawArrow">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="drawCoordinateSystem">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="drawEllipse">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="drawRect">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="drawSpline">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="eraser">      <attribute name="drawingType" type="string" value="default"/>      <attribute name="size" type="string" value="MEDIUM"/>      <attribute name="type" type="string" value="deleteStroke"/>    </data>    <data name="hand">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="highlighter">      <attribute name="color" type="hex" value="ffff00"/>      <attribute name="drawingType" type="string" value="default"/>      <attribute name="fill" type="int" value="0"/>      <attribute name="fillAlpha" type="int" value="128"/>      <attribute name="size" type="string" value="MEDIUM"/>    </data>    <data name="image">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="pen">      <attribute name="color" type="hex" value="00c0ff"/>      <attribute name="drawingType" type="string" value="strokeRecognizer"/>      <attribute name="fill" type="int" value="0"/>      <attribute name="fillAlpha" type="int" value="128"/>      <attribute name="size" type="string" value="MEDIUM"/>      <attribute name="style" type="string" value="plain"/>    </data>    <data name="playObject">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="selectObject">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="selectRect">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="selectRegion">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="showFloatingToolbox">      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="text">      <attribute name="color" type="hex" value="000000"/>      <attribute name="drawingType" type="string" value="default"/>    </data>    <data name="verticalSpace">      <attribute name="drawingType" type="string" value="default"/>    </data>  </data>  <data name="touch">    <attribute name="cmdDisable" type="string" value=""/>    <attribute name="cmdEnable" type="string" value=""/>    <attribute name="disableTouch" type="boolean" value="false"/>    <attribute name="method" type="string" value="auto"/>    <attribute name="timeout" type="int" value="1000"/>  </data></settings>


    On Wednesday, June 9, 2021, 01:02:25 PM GMT+7, Roland Lötscher ***@***.***> wrote:  
 
 


I still don't see the file. You can check in the Preview, if the file is attached.

**Comment by rolandlo** [comment]
I didn't see a screenshot, nor a file attached, but I can read the pasted content. The relevant line is
```xml
<property name="lastSavePath" value="C:\Users\charn\OneDrive\Desktop\Charming Edunicornâ„¢\#####MATH NOTE\Xournalpp Save"/> 
```
which looks suspicious.

**Comment by chanemaster** [comment]
I have folder name Charming Edunicorn and #####MATH NOTE in my computer and now I set to save Xournal Files there in folder Xournalpp. Maybe this is notmal?

ส่งจาก Yahoo Mail บน Android 
 
  เมื่อ พ., มิ.ย. 9, 2021 เวลา 13:21, Roland ***@***.***> เขียน:   


I didn't see a screenshot, nor a file attached, but I can read the pasted content. The relevant line is
<property name="lastSavePath" value="C:\Users\charn\OneDrive\Desktop\Charming Edunicornâ„¢\#####MATH NOTE\Xournalpp Save"/> 
which looks suspicious.

**Comment by rolandlo** [comment]
Most normal seems to save .xopp-files somewhere inside the Documents folder under Windows (home folder under Linux), but as long you don't run into trouble, you can save them wherever you want. I will close this, as the issue seems solved.

**Comment by chanemaster** [comment]
Most normal seems to save .xopp-files somewhere inside the Documents folder under Windows (home folder under Linux), but as long you don't run into trouble, you can save them wherever you want. I will close this, as the issue seems solved.

I think because I changed its folder to save.

Thank you for your kind assistance.

Best Regards,

Charn

ส่งจาก Yahoo Mail บน Android 
 
  เมื่อ พ., มิ.ย. 9, 2021 เวลา 14:17, Roland ***@***.***> เขียน:   


Closed #3145.
